RFC-010-Omni-Layer.md

```diff
@@ -0,0 +1,120 @@
+# The Omni Layer Asset
+
+- RFC-Number: 010
+- Status: Draft
+- Discussion-issue: -
+- Created on: 27 Mar. 2019
+
+## Table of Contents
+
+<!-- toc -->
+
+  * [Description](#description)
+  * [The Omni Layer Assets](#the-omni-layer-assets)
+    + [`quantity`](#quantity)
+    + [`property_id`](#property_id)
+  * [Registry extension](#registry-extension)
+- [Examples](#examples)
+
+<!-- tocstop -->
+
+## Description
+
+This RFC defines how the Bitcoin-based Omni Layer Asset MUST be described in Asset type headers.
+Omni Layer refers specifically to the [Omni Layer Protocol](https://github.com/OmniLayer/spec).
+
+The Asset type header was introduced in [RFC002](./RFC-002-SWAP.md) to describe assets being exchanged in a COMIT SWAP protocol.
+
+This in combination with subsequent RFCs that define concrete execution steps for particular SWAP protocols enables users to negotiate and execute Omni Layer asset swaps.
+
+## The Omni Layer Assets
+
+The Omni Layer Protocol defines several _assets_, also commonly referred as _coins_ or _tokens_ (from now on referred as Omni _asset_).
+It also allows the definition of new assets by the mean of a Bitcoin transaction containing specific data.
+
+This asset MUST be used on the Bitcoin Ledger, as defined in [RFC-004](./RFC-004-SWAP-Bitcoin.md).
+Subsequent RFCs MAY introduce the support of other ledgers for Omni Layer Assets.
+
+While *Omni* or *OmniCoin* holds a special place among the assets as being predefined and having special properties in the Omni Layer Protocol, it is not different than other Omni assets in the context of the COMIT protocol family.
+
+Each Omni asset defines its name.
+Omni assets are uniquely identified by a *property id*.
+Notable Omni assets have the following property ids:
+- Omni Coin (predefined): `1`
+- Test Omni Coin (predefined): `2`
+- TetherUS: `31`
```

Comment:
As an implementer, it's not clear what I have to do with this information. Should we be listing in a table or all the allowed `property_id`s or is this like ERC20 where we negotiate based on a the contract address and don't care about particular currencies. If it's the latter I think remove this.